bundle/edu.gemini.sp.vcs/src/main/scala/edu/gemini/sp/vcs/diff/MergePlan.scala

```diff
@@ -1,14 +1,112 @@
 package edu.gemini.sp.vcs.diff
 
-import edu.gemini.pot.sp.ISPProgram
+import edu.gemini.pot.sp._
+import edu.gemini.pot.sp.version.{VersionMap, NodeVersions}
+import edu.gemini.sp.vcs.diff.NodeDetail.Obs
+import edu.gemini.sp.vcs.diff.VcsFailure._
+import edu.gemini.spModel.rich.pot.sp._
 
 import scalaz._
+import Scalaz._
 
 /** Describes the modifications required for a local program to complete a
   * merge.
   */
 case class MergePlan(update: Tree[MergeNode], delete: Set[Missing]) {
 
+  /** "Encode" for serialization. The issue is that `scalaz.Tree` is not
+    * `Serializable` but we need to send `MergePlan`s over `trpc`.
+    */
+  def encode: MergePlan.Transport = {
+    def encodeTree(t: Tree[MergeNode]): MergePlan.TreeTransport =
+      MergePlan.TreeTransport(t.rootLabel, t.subForest.toList.map(encodeTree))
```

Comment:
It seems `subForest` is a `Stream[Tree[A]]`, would it be more performant not to materialize the stream and try computing on the stream as much as possible?